stetho/src/main/java/com/facebook/stetho/inspector/elements/android/ViewHighlightOverlays.java

```diff
@@ -0,0 +1,262 @@
+
+  abstract void highlightView(View view, int mainColor);
+
+  abstract void removeHighlight(View view);
+
+  static ViewHighlightOverlays newInstance() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2) {
+      return new ViewHighlightOverlaysJellybeanMR2();
+    }
+    return new NoOpViewHighlightOverlays();
+  }
+
+  private static class NoOpViewHighlightOverlays extends ViewHighlightOverlays {
+
+    @Override
+    void highlightView(View view, int mainColor) {
+    }
+
+    @Override
+    void removeHighlight(View view) {
+    }
+  }
+
+  @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
+  private static class ViewHighlightOverlaysJellybeanMR2 extends ViewHighlightOverlays {
+    private static final int MARGIN_OVERLAY_COLOR = 0xaaf4ca9e;
+    private static final int PADDING_OVERLAY_COLOR = 0xaabedab6;
+    private HighlightDrawable[] mHighlightDrawables = {
+        new MainHighlightDrawable(),
+        new PaddingTopHighlightDrawable(),
+        new PaddingBottomHighlightDrawable(),
+        new PaddingRightHighlightDrawable(),
+        new PaddingLeftHighlightDrawable(),
+        new MarginTopHighlightDrawable(),
+        new MarginBottomHighlightDrawable(),
+        new MarginRightHighlightDrawable(),
+        new MarginLeftHighlightDrawable()
+    };
+
+    ViewHighlightOverlaysJellybeanMR2() {
+    }
+
+    @Override
+    void highlightView(View view, int mainColor) {
+      mHighlightDrawables[0].setColor(mainColor);
+
+      for (HighlightDrawable mHighlightDrawable : mHighlightDrawables) {
+        mHighlightDrawable.highlightView(view);
+      }
+    }
+
+    @Override
+    void removeHighlight(View view) {
+      for (ColorDrawable drawable : mHighlightDrawables) {
+        view.getOverlay().remove(drawable);
+      }
+    }
+
+    abstract class HighlightDrawable extends ColorDrawable {
+
+      Rect mMargins = new Rect();
+      Rect mPaddings = new Rect();
+
+      HighlightDrawable(int color) {
+        super(color);
+      }
+
+      public HighlightDrawable() {
+      }
+
+      void highlightView(View view) {
+        ViewGroup.LayoutParams layoutParams = view.getLayoutParams();
+        if (layoutParams instanceof MarginLayoutParams) {
+          MarginLayoutParams marginLayoutParams = (MarginLayoutParams) layoutParams;
+          mMargins = new Rect(marginLayoutParams.leftMargin, marginLayoutParams.topMargin,
+              marginLayoutParams.rightMargin, marginLayoutParams.bottomMargin);
+        }
+        mPaddings = new Rect(view.getPaddingLeft(), view.getPaddingTop(), view.getPaddingRight(),
+            view.getPaddingBottom());
+      }
+    }
+
+    class MainHighlightDrawable extends HighlightDrawable {
+
+      @Override
+      void highlightView(View view) {
+        super.highlightView(view);
+        setBounds(0, 0, view.getWidth(), view.getHeight());
+        view.getOverlay().add(this);
+      }
+
+      @Override
+      public void draw(@NonNull Canvas canvas) {
```

Comment:
Android Studio gives me an warning asking to add this annotation, but I can remove if you don't care :)